### DISCUSSION TITLE
Compiling of my own projects in VS2017 are very slow

### DISCUSSION THREAD

**Discussion by 330305020** [gabime/spdlog#2829] [Q&A]
I used VS2017 and have my own projects in solution. In projects, I added #include "spdlog/spdlog.h", and it leads that compiling of projects takes very very long time. Do you know how to solve it? Thanks.

**Comment by tt4g** [comment]
In general, using a header-only library will require more time to build the project.
Using pre-compiled libraries should reduce build time.

For CMake projects, `find_package(spdlog)` (if spdlog is installed on your machine) or `add_subdirectory()` (to make spdlog part of your project) will add the `spdlog::spdlog` target to your project so `target_link_libraries(<your_target> spdlog::spdlog)` will add the compiled libraries to the project.

See also: #2542. #2528

**Reply by 330305020** [reply]
Thanks, now I know what to do. Before I already have compiled lib, and what I only need to do is to add SPDLOG_COMPILED_LIB and SPDLOG_SHARED_LIB to the preprocessor of my project. Then it works.
